extensions/wikia/CuratedContent/CuratedContentValidator.class.php

```diff
@@ -14,188 +15,170 @@ class CuratedContentValidator {
 	const ERR_EMPTY_LABEL = 'emptyLabel';
 	const ERR_NO_CATEGORY_IN_TAG = 'noCategoryInTag';
 
-	private $errors;
-	private $existingSectionLabels;
-	private $existingItemLabels;
-	private $existingFeaturedItemLabels;
-	private $hasOptionalSection;
-
-	public function __construct() {
-		$this->reset();
-	}
-
-	public function reset() {
-		$this->errors = [ ];
-		$this->existingSectionLabels = [ ];
-		$this->existingItemLabels = [ ];
-		$this->existingFeaturedItemLabels = [ ];
-		$this->hasOptionalSection = false;
-	}
-
 	public function validateData( $data ) {
-		$this->reset();
-
-		// validate sections
-		foreach ( $data as $section ) {
-			if ( !empty( $section['featured'] ) ) {
-				$this->validateItems( $section, true );
-			} else {
-				$this->validateSection( $section );
-				$this->validateItemsExist( $section );
-				$this->validateItems( $section );
-				$this->validateItemsTypes( $section );
+		$errors = [ ];
+		$alreadyUsedSectionLabels = [ ];
+
+		if ( is_array( $data ) ) {
+			foreach ( $data as $section ) {
+
+				if ( is_array( $section['items'] ) ) {
+					if ( !empty( $section['featured'] ) ) {
+						foreach ( $section['items'] as $featuredItem ) {
+							if ( !empty( $this->validateFeaturedItem( $featuredItem ) ) ) {
+								$errors[] = self::ERR_OTHER_ERROR;
```

Comment:
let's revisit this idea when we are going to fix multiple error notifications issue.
If we would like to return false at the beginning we might loose information about duplicated labels in sections